btk/survey.py

```diff
@@ -56,90 +65,91 @@
     """
-    OmegaConf.resolve(survey_conf)  # in-place
-
-    survey_dict = dict(survey_conf)
-    filters_dict = dict(survey_dict.pop("filters"))
-    filter_names = list(filters_dict.keys())
-    for key in filter_names:
-        filter_dict = dict(filters_dict.pop(key))  # need to make it a dict not DictConfig
-        psf_dict = dict(filter_dict.pop("psf"))
-        if "type" not in psf_dict:
-            raise AttributeError(
-                f"Your configuration for the PSF in filter {key} in survey {survey_dict['name']}"
-                f"does not have a 'type' field which is required."
-            )
-        if psf_dict["type"] == "default":
-            psf = get_psf(**psf_dict["params"])
-        elif psf_dict["type"] == "galsim":
-            galsim_dict = dict(psf=psf_dict["params"])
-            psf = galsim.config.BuildGSObject(galsim_dict, "psf")
-        else:
-            raise NotImplementedError(
-                f"The 'type' specified for the PSF in filter {key} in "
-                f"survey {survey_dict['name']} is not implemented."
-            )
+    if isinstance(names, str):
+        names = [names]
+    if not isinstance(names, list):
+        raise TypeError("Argument 'names' of `get_surveys` should be a str or list.")
 
-        filter_dict["psf"] = psf
-        filters_dict[key] = filter_dict
-    filters = [Filter(**filters_dict[key]) for key in filters_dict]
-    survey = Survey(**survey_dict, filters=filters)
-    return survey
+    btk_surveys = []
+    for survey_name in names:
+        survey = galcheat.get_survey(survey_name)
+        filters = []
+        for band, filtr in survey.get_filters().items():
+            if psf_func is None:
+                psf = get_default_galcheat_psf(survey, filtr)
+            else:
+                psf = psf_func(survey, filtr)
+            btk_filter = Filter(
+                band,
+                psf,
+                filtr.sky_brightness.value,
+                filtr.exposure_time.value,
+                filtr.zeropoint.value,
+                filtr.extinction.value,
+            )
+            filters.append(btk_filter)
+
+        btk_survey = Survey(
+            survey.name,
+            survey.pixel_scale.value,
+            survey.effective_area.value,
+            survey.mirror_diameter.value,
+            survey.airmass.value,
+            survey.zeropoint_airmass.value,
+            filters,
+        )
+        btk_surveys.append(btk_survey)
 
+    if len(btk_surveys) == 1:
+        return btk_surveys[0]
+    return btk_surveys
 
-def get_surveys(names="Rubin", overrides: Iterable = ()):
-    """Return specified surveys as `btk.survey.Survey` objects.
 
-    NOTE: The surveys currently implemented correspond to config files inside `conf/surveys`. See
-    the documentation for how to add your own surveys via custom config files.
+def get_default_galcheat_psf(survey: galcheat.survey.Survey, filtr: galcheat.filter.Filter):
```

Comment:
I find the name `galcheat_psf` confusing because it kind of suggests galcheat contains a PSF.

Reply:
good point, perhaps `get_default_psf` was better